solidity/libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -823,11 +823,19 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 		case FunctionType::Kind::ECRecover:
 		case FunctionType::Kind::SHA256:
 		case FunctionType::Kind::RIPEMD160:
+		case FunctionType::Kind::EDVerify:
+		case FunctionType::Kind::BLAKE2b256:
+		case FunctionType::Kind::TXHash:
 		{
 			_functionCall.expression().accept(*this);
-			static const map<FunctionType::Kind, u128> contractAddresses{{FunctionType::Kind::ECRecover, 1},
-															   {FunctionType::Kind::SHA256, 2},
-															   {FunctionType::Kind::RIPEMD160, 3}};
+			static const map<FunctionType::Kind, u128> contractAddresses{
+				{FunctionType::Kind::ECRecover, 1},
+				{FunctionType::Kind::SHA256, 2},
+				{FunctionType::Kind::RIPEMD160, 3},
+				{FunctionType::Kind::EDVerify, 16},
```

Comment:
What do those numbers mean?  (16, 17, 18)

Reply:
the contact address offset will be present in the Aion precompile contract